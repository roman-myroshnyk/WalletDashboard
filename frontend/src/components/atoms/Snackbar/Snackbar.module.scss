@import '@/styles/colors.scss';
@import '@/styles/mixins.scss';

.li {
  position: relative;

  display: flex;
  align-items: center;
  justify-content: space-between;
  max-width: 100%;
  min-height: 60px;
  margin-top: 8px;
  margin-right: 14px;
  margin-bottom: 8px;
  padding: 0;
  padding-right: 10px;
  padding-left: 10px;

  background: white;
  background-color: $darkBlack;
  border: 2px solid $lightBlack;
  border-radius: 10px;

  pointer-events: auto;

  @include media-breakpoint-up(sm) {
    max-width: 50%;
    margin-right: 38px;
    margin-left: 38px;
  }
}

.warningIcon {
  display: flex;
  align-items: center;
  justify-content: center;
  padding-right: 12px;
  padding-left: 12px;

  color: $red;
}

.text {
  padding-right: 10px;
  padding-left: 10px;

  color: $gray;
  font-weight: 500;
  font-size: 18px;
  font-family: Epilogue, sans-serif;
  font-style: normal;
  line-height: 24px;

  list-style: none;
}
