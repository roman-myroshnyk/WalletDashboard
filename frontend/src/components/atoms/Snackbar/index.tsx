import { FC } from 'react';
import { motion } from 'framer-motion';
// components
import WarningIcon from '@/icons/WarningIcon';
import IconButton from '@/atoms/IconButton';
import CancelIcon from '@/icons/CancelIcon';
// styles
import { animations } from '@/animations/SnackbarAnimations';
import styles from './Snackbar.module.scss';

interface IProps {
    onClose: () => void;
    message: string;
    snackbarKey:string;
}

const Snackbar:FC<IProps> = ({
  message,
  onClose,
  snackbarKey,
}) => (
  <motion.li
    className={styles.li}
    layout
    variants={animations}
  >
    <span className={styles.warningIcon}>
      <WarningIcon />
    </span>
    <span className={styles.text}>
      {message}
      {snackbarKey}
    </span>
    <IconButton
      label="cancel"
      isShown
      disabled={false}
      onClick={onClose}
    >
      <CancelIcon
        label="cancel"
        enterDelay={1}
        exitDelay={1}
      />
    </IconButton>
  </motion.li>
);

export default Snackbar;
