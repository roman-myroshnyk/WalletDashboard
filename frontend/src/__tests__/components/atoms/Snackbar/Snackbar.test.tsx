import '.../../../mocks/framer-motion';
import { render, fireEvent } from '@testing-library/react';
import Snackbar from '@/atoms/Snackbar';

describe('atoms/Snackbar', () => {
  it('Snackbar should render', () => {
    const closeHandler = jest.fn();
    const { container } = render(
      <Snackbar message="I am snackbar message" onClose={closeHandler} />,
    );
    expect(container).toMatchSnapshot();
  });

  it('Snackbar should respond on click events', () => {
    const closeHandler = jest.fn();
    const { getByLabelText } = render(
      <Snackbar message="I am snackbar message" onClose={closeHandler} />,
    );
    const cancelButton = getByLabelText('cancel');
    fireEvent.click(cancelButton);
    expect(closeHandler).toBeCalledTimes(1);
  });
});
