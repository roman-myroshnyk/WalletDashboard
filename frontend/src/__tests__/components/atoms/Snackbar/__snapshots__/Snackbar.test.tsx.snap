// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/Snackbar Snackbar should render 1`] = `
<div>
  <span
    class="warningIcon"
  >
    <div
      style="display: block; align-items: center; justify-content: center; width: 24px; height: 24px;"
      transition="[object Object]"
    >
      <svg
        aria-label="warningIcon"
        class="h-5 w-5"
        fill="currentColor"
        height="24"
        viewBox="0 0 20 20"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clip-rule="evenodd"
          d="M8.257 3.099c.765-1.36 2.722-1.36 3.486 0l5.58 9.92c.75 1.334-.213 2.98-1.742 2.98H4.42c-1.53 0-2.493-1.646-1.743-2.98l5.58-9.92zM11 13a1 1 0 11-2 0 1 1 0 012 0zm-1-8a1 1 0 00-1 1v3a1 1 0 002 0V6a1 1 0 00-1-1z"
          fill="currentColor"
          fill-rule="evenodd"
        />
      </svg>
    </div>
  </span>
  <span
    class="text"
  >
    I am snackbar message
  </span>
  <div>
    <button
      area-label="cancel"
      class="iconButton"
      variants="[object Object]"
    >
      <svg
        animate="enter"
        aria-label="cancel"
        class="h-6 w-6"
        exit="exit"
        fill="none"
        height="24"
        initial="exit"
        stroke="currentColor"
        stroke-width="2"
        variants="[object Object]"
        viewBox="0 0 24 24"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M6 18L18 6M6 12"
          stroke-linecap="round"
          stroke-linejoin="round"
          variants="[object Object]"
        />
        <path
          d="M6 6M6 6l12 12"
          stroke-linecap="round"
          stroke-linejoin="round"
          variants="[object Object]"
        />
      </svg>
    </button>
  </div>
</div>
`;
